relay/transformer/transformer.go

```diff
@@ -0,0 +1,100 @@
+// Copyright 2019 PingCAP, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package transformer
+
+import (
+	"github.com/pingcap/parser"
+	"github.com/siddontang/go-mysql/mysql"
+	"github.com/siddontang/go-mysql/replication"
+)
+
+var (
+	artificialFlag = uint16(0x0020) // LOG_EVENT_ARTIFICIAL_F
+)
+
+// Result represents a transform result.
+type Result struct {
```

Comment:
I don't understand this design very well,  like
* if some binlog is ignored, should writer write it?
* what's its transform rules?

Reply:
binlog event may be ignored in multiple steps in the processing chain, at least:
1. some fake/artificial events, which will be marked as ignored in this transformer
2. some obsolete events (like FormatDescriptionEvent), which will be ignored by the writer

maybe the name of `transformer` is not accurate, and in fact, more features may be added into this later.